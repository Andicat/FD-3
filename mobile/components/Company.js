import React from 'react';
import PropTypes from 'prop-types';
import CompanyEvents from './events';

import Client from './Client';
import ClientCard from './ClientCard';

class Company extends React.PureComponent {

    static propTypes = {
        name: PropTypes.string.isRequired,
        clients:PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                lastName: PropTypes.string.isRequired,
                name: PropTypes.string.isRequired,
                midName: PropTypes.string.isRequired,
                balance: PropTypes.number.isRequired,
                activity: PropTypes.string.isRequired,
            })
        ),
    };

    static defaultProps = {
        name: 'Velcom',
        clients: [],
    };

    state = {
        name: this.props.name,
        showActive: '',
        clientToEdit: null,
        clientsToShow:this.props.clients.slice(),
    }

    setCompanyName = (name) => {
        this.setState({name:name});
    }

    setFilter = (activity) => {
        this.setState({showActive:activity});
    }

    componentDidMount = () => {
        CompanyEvents.addListener('deleteClient',this.deleteClient);
        CompanyEvents.addListener('editClient',this.editClient);
        CompanyEvents.addListener('updateClient',this.updateClient);
    }

    componentWillUnmount = () => {
        CompanyEvents.removeListener('deleteClient',this.deleteClient);
        CompanyEvents.removeListener('editClient',this.editClient);
        CompanyEvents.removeListener('updateClient',this.updateClient);
    }

    deleteClient = (clientId) => {
        this.setState({clientsToShow:this.state.clientsToShow.filter(c => c.id!=clientId)});
    };

    editClient = (client) => {
        this.setState({clientToEdit:null}, () => {this.setState({clientToEdit:client})});
    };

    updateClient = (clientNew) => {
        let clientsToShowNew;
        if (clientNew.id) {
            clientsToShowNew = this.state.clientsToShow.map(c => c.id!=clientNew.id?c:clientNew);
        } else {
            let newId = this.state.clientsToShow.reduce(function (r, v) { return ( r < v.id ? v.id : r);},0) + 1;
            clientsToShowNew = [...this.state.clientsToShow,{id:newId,...clientNew}];
        }
        this.setState({clientsToShow:clientsToShowNew,clientToEdit:null});
    };

    newClient = () => {
        this.setState( {clientToEdit:{lastName: "", name: "", midName: "", balance: 0, activity: "active",}} );
    }

    render() {
        console.log(`Company ${this.state.name} render`);
    
        var clientsTableTitleCode = 
            <div className='company__table-title'>
                <span className='titleLastName'>Фамилия</span>
                <span className='titleName'>Имя</span>
                <span className='titleMIdName'>Отчество</span>
                <span className='titleBalance'>Баланс</span>
                <span className='titleActivity'>Активность</span>
                <span className='titleEdit'>Редактировать</span>
                <span className='titleDelete'>Удалить</span>
            </div>;

        var clientsCode=this.state.clientsToShow.map( c => (this.state.showActive&&this.state.showActive!=c.activity)?null:<Client key={c.id} client={c}/>);

        if (this.state.clientToEdit) {
            var cardCode = <ClientCard client={this.state.clientToEdit}/>;
        };

        return (
            <div className='company'>
                <div className='company__name'>
                    <input type="button" value="МТС" onClick={this.setCompanyName.bind(this,"МТС")}/>
                    <input type="button" value="Velcom" onClick={this.setCompanyName.bind(this,"Velcom")}/>
                    <h2>Компания: {this.state.name}</h2>
                </div>
                <div className='company__filter'>
                    <input type="button" value="Все" onClick={this.setFilter.bind(this,"")} />
                    <input type="button" value="Активные" onClick={this.setFilter.bind(this,"active")} />
                    <input type="button" value="Заблокированные" onClick={this.setFilter.bind(this,"blocked")} />
                </div>
                <div className='company__table'>
                    {clientsTableTitleCode}
                    {clientsCode}
                </div>
                <input className='company__add'type="button" value="Добавить клиента" onClick={this.newClient} />
                {cardCode}
            </div>
        )
    }
}

export default Company;