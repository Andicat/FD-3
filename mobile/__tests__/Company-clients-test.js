"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import Company from '../components/Company';
import CompanyEvents from '../components/events';

test('работа Company - добавление клиента', () => {

    let clientsArr = [ 
        {"id":1,"lastName":"Иванов","name":"Иван","midName":"Иванович","balance":200,"activity":"active"},
        {"id":2,"lastName":"Сидоров","name":"Сидор","midName":"Сидорович","balance":250,"activity":"active"},
        {"id":3,"lastName":"Петров","name":"Петр","midName":"Петрович","balance":180,"activity":"active"},
        {"id":4,"lastName":"Григорьев","name":"Григорий","midName":"Григорьевич","balance":-200,"activity":"blocked"}
    ]
;

    //создаём тестовую версию компонента
    const component = renderer.create(
    <Company clients={clientsArr}/>
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    //вызываем событие добавления нового клиента
    let newClientData = {"lastName":"Новый","name":"Клиент","midName":"Иванович","balance":100,"activity":"active"};
    CompanyEvents.emit('updateClient',newClientData);
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    //вызываем событие редактирования клиента
    let upClientData = {"id":3,"lastName":"Новый","name":"Клиент","midName":"Иванович","balance":100,"activity":"active"};
    CompanyEvents.emit('updateClient',upClientData);
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    //вызываем событие удаление клиента
    let idDelClient = 1;
    CompanyEvents.emit('deleteClient',idDelClient);
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    //вызываем событие открытия карточки клиента
    let clientToEdit = clientsArr[1];
    CompanyEvents.emit('editClient',clientToEdit);
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});
