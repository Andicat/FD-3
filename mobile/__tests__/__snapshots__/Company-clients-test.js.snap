// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Company - добавление клиента 1`] = `
<div
  className="company"
>
  <div
    className="company__name"
  >
    <input
      onClick={[Function]}
      type="button"
      value="МТС"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="company__filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div
    className="company__table"
  >
    <div
      className="company__table-title"
    >
      <span
        className="titleLastName"
      >
        Фамилия
      </span>
      <span
        className="titleName"
      >
        Имя
      </span>
      <span
        className="titleMIdName"
      >
        Отчество
      </span>
      <span
        className="titleBalance"
      >
        Баланс
      </span>
      <span
        className="titleActivity"
      >
        Активность
      </span>
      <span
        className="titleEdit"
      >
        Редактировать
      </span>
      <span
        className="titleDelete"
      >
        Удалить
      </span>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Иванов
      </span>
      <span
        className="client__name"
      >
        Иван
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        200
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Сидоров
      </span>
      <span
        className="client__name"
      >
        Сидор
      </span>
      <span
        className="client__midName"
      >
        Сидорович
      </span>
      <span
        className="client__balance"
      >
        250
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Петров
      </span>
      <span
        className="client__name"
      >
        Петр
      </span>
      <span
        className="client__midName"
      >
        Петрович
      </span>
      <span
        className="client__balance"
      >
        180
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--blocked"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Григорьев
      </span>
      <span
        className="client__name"
      >
        Григорий
      </span>
      <span
        className="client__midName"
      >
        Григорьевич
      </span>
      <span
        className="client__balance"
      >
        -200
      </span>
      <span
        className="client__activity"
      >
        blocked
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
  </div>
  <input
    className="company__add"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа Company - добавление клиента 2`] = `
<div
  className="company"
>
  <div
    className="company__name"
  >
    <input
      onClick={[Function]}
      type="button"
      value="МТС"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="company__filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div
    className="company__table"
  >
    <div
      className="company__table-title"
    >
      <span
        className="titleLastName"
      >
        Фамилия
      </span>
      <span
        className="titleName"
      >
        Имя
      </span>
      <span
        className="titleMIdName"
      >
        Отчество
      </span>
      <span
        className="titleBalance"
      >
        Баланс
      </span>
      <span
        className="titleActivity"
      >
        Активность
      </span>
      <span
        className="titleEdit"
      >
        Редактировать
      </span>
      <span
        className="titleDelete"
      >
        Удалить
      </span>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Иванов
      </span>
      <span
        className="client__name"
      >
        Иван
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        200
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Сидоров
      </span>
      <span
        className="client__name"
      >
        Сидор
      </span>
      <span
        className="client__midName"
      >
        Сидорович
      </span>
      <span
        className="client__balance"
      >
        250
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Петров
      </span>
      <span
        className="client__name"
      >
        Петр
      </span>
      <span
        className="client__midName"
      >
        Петрович
      </span>
      <span
        className="client__balance"
      >
        180
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--blocked"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Григорьев
      </span>
      <span
        className="client__name"
      >
        Григорий
      </span>
      <span
        className="client__midName"
      >
        Григорьевич
      </span>
      <span
        className="client__balance"
      >
        -200
      </span>
      <span
        className="client__activity"
      >
        blocked
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Новый
      </span>
      <span
        className="client__name"
      >
        Клиент
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        100
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
  </div>
  <input
    className="company__add"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа Company - добавление клиента 3`] = `
<div
  className="company"
>
  <div
    className="company__name"
  >
    <input
      onClick={[Function]}
      type="button"
      value="МТС"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="company__filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div
    className="company__table"
  >
    <div
      className="company__table-title"
    >
      <span
        className="titleLastName"
      >
        Фамилия
      </span>
      <span
        className="titleName"
      >
        Имя
      </span>
      <span
        className="titleMIdName"
      >
        Отчество
      </span>
      <span
        className="titleBalance"
      >
        Баланс
      </span>
      <span
        className="titleActivity"
      >
        Активность
      </span>
      <span
        className="titleEdit"
      >
        Редактировать
      </span>
      <span
        className="titleDelete"
      >
        Удалить
      </span>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Иванов
      </span>
      <span
        className="client__name"
      >
        Иван
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        200
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Сидоров
      </span>
      <span
        className="client__name"
      >
        Сидор
      </span>
      <span
        className="client__midName"
      >
        Сидорович
      </span>
      <span
        className="client__balance"
      >
        250
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Новый
      </span>
      <span
        className="client__name"
      >
        Клиент
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        100
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--blocked"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Григорьев
      </span>
      <span
        className="client__name"
      >
        Григорий
      </span>
      <span
        className="client__midName"
      >
        Григорьевич
      </span>
      <span
        className="client__balance"
      >
        -200
      </span>
      <span
        className="client__activity"
      >
        blocked
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Новый
      </span>
      <span
        className="client__name"
      >
        Клиент
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        100
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
  </div>
  <input
    className="company__add"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа Company - добавление клиента 4`] = `
<div
  className="company"
>
  <div
    className="company__name"
  >
    <input
      onClick={[Function]}
      type="button"
      value="МТС"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="company__filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div
    className="company__table"
  >
    <div
      className="company__table-title"
    >
      <span
        className="titleLastName"
      >
        Фамилия
      </span>
      <span
        className="titleName"
      >
        Имя
      </span>
      <span
        className="titleMIdName"
      >
        Отчество
      </span>
      <span
        className="titleBalance"
      >
        Баланс
      </span>
      <span
        className="titleActivity"
      >
        Активность
      </span>
      <span
        className="titleEdit"
      >
        Редактировать
      </span>
      <span
        className="titleDelete"
      >
        Удалить
      </span>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Сидоров
      </span>
      <span
        className="client__name"
      >
        Сидор
      </span>
      <span
        className="client__midName"
      >
        Сидорович
      </span>
      <span
        className="client__balance"
      >
        250
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Новый
      </span>
      <span
        className="client__name"
      >
        Клиент
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        100
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--blocked"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Григорьев
      </span>
      <span
        className="client__name"
      >
        Григорий
      </span>
      <span
        className="client__midName"
      >
        Григорьевич
      </span>
      <span
        className="client__balance"
      >
        -200
      </span>
      <span
        className="client__activity"
      >
        blocked
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Новый
      </span>
      <span
        className="client__name"
      >
        Клиент
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        100
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
  </div>
  <input
    className="company__add"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа Company - добавление клиента 5`] = `
<div
  className="company"
>
  <div
    className="company__name"
  >
    <input
      onClick={[Function]}
      type="button"
      value="МТС"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="company__filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div
    className="company__table"
  >
    <div
      className="company__table-title"
    >
      <span
        className="titleLastName"
      >
        Фамилия
      </span>
      <span
        className="titleName"
      >
        Имя
      </span>
      <span
        className="titleMIdName"
      >
        Отчество
      </span>
      <span
        className="titleBalance"
      >
        Баланс
      </span>
      <span
        className="titleActivity"
      >
        Активность
      </span>
      <span
        className="titleEdit"
      >
        Редактировать
      </span>
      <span
        className="titleDelete"
      >
        Удалить
      </span>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Сидоров
      </span>
      <span
        className="client__name"
      >
        Сидор
      </span>
      <span
        className="client__midName"
      >
        Сидорович
      </span>
      <span
        className="client__balance"
      >
        250
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Новый
      </span>
      <span
        className="client__name"
      >
        Клиент
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        100
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--blocked"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Григорьев
      </span>
      <span
        className="client__name"
      >
        Григорий
      </span>
      <span
        className="client__midName"
      >
        Григорьевич
      </span>
      <span
        className="client__balance"
      >
        -200
      </span>
      <span
        className="client__activity"
      >
        blocked
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
    <div
      className="client client--active"
      onClick={undefined}
    >
      <span
        className="client__lastName"
      >
        Новый
      </span>
      <span
        className="client__name"
      >
        Клиент
      </span>
      <span
        className="client__midName"
      >
        Иванович
      </span>
      <span
        className="client__balance"
      >
        100
      </span>
      <span
        className="client__activity"
      >
        active
      </span>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Редактировать
      </button>
      <button
        className="client__button"
        onClick={[Function]}
        type="button"
      >
        Удалить
      </button>
    </div>
  </div>
  <input
    className="company__add"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <div
    className="clientCard"
  >
    <div
      className="clientCard__field"
    >
      <label
        htmlFor="lastName"
      >
        Фамилия:
      </label>
      <input
        defaultValue="Сидоров"
        id="lastName"
        name="lastName"
        type="text"
      />
    </div>
    <div
      className="clientCard__field"
    >
      <label
        htmlFor="name"
      >
        Имя:
      </label>
      <input
        defaultValue="Сидор"
        id="name"
        name="name"
        type="text"
      />
    </div>
    <div
      className="clientCard__field"
    >
      <label
        htmlFor="midName"
      >
        Отчество:
      </label>
      <input
        defaultValue="Сидорович"
        id="midName"
        name="midName"
        type="text"
      />
    </div>
    <div
      className="clientCard__field"
    >
      <label
        htmlFor="balance"
      >
        Баланс:
      </label>
      <input
        defaultValue={250}
        id="balance"
        name="balance"
        type="text"
      />
    </div>
    <div
      className="clientCard__field"
    >
      <button
        className="clientCard__save"
        onClick={[Function]}
        type="button"
      >
        Сохранить 
      </button>
      <button
        className="clientCard__cancel"
        onClick={[Function]}
        type="button"
      >
        Отмена
      </button>
    </div>
  </div>
</div>
`;
